security/tinc/src/opnsense/mvc/app/models/OPNsense/Tinc/Tinc.xml

```diff
@@ -123,7 +131,7 @@
                     <mask>/^([0-9a-zA-Z\.,_\-:]){0,1024}$/u</mask>
                 </extaddress>
                 <subnet type="NetworkField">
-                    <Required>Y</Required>
+                    <Required>N</Required>
```

Comment:
Isn't this related to the same mode field?

Reply:
You are right. This field (host.subnet) should be set when the network such a host is assigned to (host.network) is in switch mode (network.mode). But I'm not sure how to implement a cross-model constraint and https://docs.opnsense.org/development/frontend/models_constraints.html has nothing about it. Could you advise me on any relevant example to follow?

Reply:
you're right, it can't know that since it has to read the relation field. you can hook into the validation by extending `performValidation()`, but maybe that's a too large a change to add now. for inspiration, you could look at https://github.com/opnsense/plugins/blob/master/net/firewall/src/opnsense/mvc/app/models/OPNsense/Firewall/Filter.php

Reply:
It's a good idea, I understand the approach but I think I'm not ready yet to dive so deep now. Nevertheless, I will keep it in mind.